src/modals/ConnectModal.svelte

```diff
@@ -0,0 +1,180 @@
+<script lang="ts">
+  import * as httpd from "@app/lib/httpd";
+  import * as modal from "@app/lib/modal";
+  import { httpdStore } from "@app/lib/httpd";
+
+  import Command from "@app/components/Command.svelte";
+  import FullscreenModal from "@app/components/FullscreenModal.svelte";
+  import Icon from "@app/components/Icon.svelte";
+  import IconButton from "@app/components/IconButton.svelte";
+  import IconSmall from "@app/components/IconSmall.svelte";
+  import Link from "@app/components/Link.svelte";
+  import TextInput from "@app/components/TextInput.svelte";
+
+  export let title: string;
+  export let subtitle: string[];
+
+  $: customUrl = `${httpd.api.baseUrl.scheme}://${httpd.api.baseUrl.hostname}:${customPort}`;
+  $: command = import.meta.env.PROD
+    ? `rad web --backend ${customUrl}`
+    : `rad web --frontend ${
+        new URL(import.meta.url).origin
+      } --backend ${customUrl}`;
+  let customPort = httpd.api.port;
+  $: validPortNumber = Number(customPort) > 0 && Number(customPort) <= 65535;
+
+  $: if ($httpdStore.state === "authenticated") {
+    modal.hide();
+  }
+</script>
+
+<style>
+  .container {
+    display: flex;
+    flex-direction: column;
+    width: 33rem;
+    margin-top: 1.5rem;
+    gap: 1.5rem;
+  }
+  .progress {
+    display: flex;
+    flex-direction: column;
+    gap: 0.5rem;
+  }
+  .progress-bar {
+    height: 6px;
+    border-radius: var(--border-radius-round);
+    background-color: var(--color-background-dip);
+  }
+  .bar {
+    display: flex;
+    background-color: var(--color-fill-secondary);
+    height: 100%;
+    border-radius: var(--border-radius-round);
+  }
+  .captions {
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr;
+    color: var(--color-foreground-dim);
+    font-size: var(--font-size-tiny);
+  }
+
+  .input {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    width: 100%;
+    gap: 0.75rem;
```

Comment:
If I interpret this correctly the only purpose of `.input` is to add spacing between `.label` and what follows it. I would recommend using margins or spacer divs like `<div style:margin-top: "0.75rem" />` and eliminating the `.input` wrapper. This simplifies the code by making the element tree more shallow and reduces the number of CSS declarations. The same applies to `.progress` and `.container`.

Reply:
When I implement things, I usually try to mimic how things are designed in figma. Figma kind-of mimics flex-box and a lot of the designs rely on the `gap` property.

<img width="672" alt="Screenshot 2023-10-12 at 14 42 55" src="https://github.com/radicle-dev/radicle-interface/assets/158411/b40722e3-8d0d-435f-b468-777747650fbe">
<img width="972" alt="Screenshot 2023-10-12 at 14 43 30" src="https://github.com/radicle-dev/radicle-interface/assets/158411/c0ebbaaf-d636-4448-b26f-2c59396b035d">

But that's besides the point. I guess we could try to make this less nested and get by with margins. However this whole connect modal should go away soon as we streamline the onboarding process. Therefore I don't want to spend more time tuning it.